tools/traffic_vault_migrate/postgres.go

```diff
@@ -0,0 +1,716 @@
+package main
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import (
+	"database/sql"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/apache/trafficcontrol/lib/go-tc"
+	util "github.com/apache/trafficcontrol/lib/go-util"
+	_ "github.com/lib/pq"
+	"log"
+	"reflect"
+	"strconv"
+	"strings"
+)
+
+type PGConfig struct {
+	Host     string `json:"host"`
+	Port     string `json:"port"`
+	User     string `json:"user"`
+	Password string `json:"password"`
+	SSLMode  string `json:"sslmode"`
+	Database string `json:"database"`
+	Key      string `json:"aesKey"`
+	AESKey   []byte
+}
+type PGBackend struct {
+	sslKey PGSSLKeyTable
+	dnssec PGDNSSecTable
+	uri    PGURISignKeyTable
+	url    PGURLSigKeyTable
+	cfg    PGConfig
+	db     *sql.DB
+}
+
+func (pg *PGBackend) String() string {
```

Comment:
GoDoc? Also, does this receiver need to be a pointer?

(Typically for `func (this Thing) String() string` methods I just write "String implements the fmt.Stringer interface.")

Reply:
I don't care about the pointer receiver thing, I meant to take that out of the comment. For a second I was like "well what if it's null?" - if that's true, then `nil.String()` is a segfault - that receiver type should never matter unless values are written.